src/shared/watchdog.c

```diff
@@ -40,7 +40,9 @@ static int update_timeout(void) {
                 if (ioctl(watchdog_fd, WDIOC_SETTIMEOUT, &sec) < 0)
                         return log_warning_errno(errno, "Failed to set timeout to %is: %m", sec);
 
-                log_info("Set hardware watchdog to %s.", FORMAT_TIMESPAN(sec * USEC_PER_SEC, 0));
+                /* watchdog_timeout stores the actual timeout used by the HW */
```

Comment:
given how shitty drivers are maybe validate the value they return. i.e. an assert() probably is sufficient, that encodes are API assumptions here:

```c
assert(sec > 0);
```

Reply:
hmm an assert would crash PID1.

Wouldn't it be better to assume that the passed timeout value has been used by the driver (since it returned 0) instead ?

Reply:
OK